### PR TITLE
[FIX] sale: distribution model used for confirmed order

### PR DESCRIPTION
**Pull request by gawa-odoo**
To reproduce:
- Create a Distribution model, with an analytic distribution and an effect on Partner A
- Create a Sale Order, for Partner A
- Confirm it
- Add a line on this Sale Order 
- => The new line does not get the distribution of the model

It couldn't be done before 16.0 because a compute would trigger when posting the sale order, because date_order was used for the default rules, and so would trigger it at wrong times.
Distribution Models don't use this date anymore, so the compute won't happen at wrong times.

task-3355886

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Feyensv** [comment]
Introduced by https://github.com/odoo/odoo/commit/04a352ea760ee8205fc2e928cbb9a7dd4c85049c, but not sure what's the expected/intended behavior, I'll let you guys decide :D

**Comment by tsb-odoo** [comment]
@william-andre LGTM functionally speaking.

File: addons/sale/models/sale_order_line.py [hunk 1]
```diff
@@ -894,7 +894,7 @@ def _compute_untaxed_amount_to_invoice(self):
     @api.depends('order_id.partner_id', 'product_id')
     def _compute_analytic_distribution(self):
         for line in self:
-            if not line.display_type and line.state == 'draft':
```
[the remark this answers is not in the document]
Comment (gawa-odoo) [reply addons/sale/models/sale_order_line.py:897]: Yes, I forgot to say it, I will edit the commit message to reflect that. Analytic accounts's compute depended on date_order for the analytic default rules, while it is not the case anymore with distribution models

**Review by antoine162 — APPROVED** [review]
To me this looks good, I'll try to get someone to r+ while JCO is away :smile:

**Review by smetl — COMMENTED** [review]
@robodoo r+

**Pull request closed without merging**